src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs

```diff
@@ -153,6 +153,15 @@ private void OnBeginExecute(EventWrittenEventArgs eventData)
                     activity.SetTag(SemanticConventions.AttributeDbStatement, commandText);
                 }
             }
+            
+            try
+            {
+                this.options.Enrich?.Invoke(activity, "OnCustom", eventData.Payload);
```

Comment:
Is payload useful to anyone? IIRC we didn't add this because it is just an array of a few strings and there was no way to get at the raw request (command object).

Reply:
Nothing I am aware of, I cannot think of any reason to use the SqlCommand either, but it encapsulates the same idea as the SqlCommand that is passed from .NET in providing the raw data from with the activity is created from.

Reply:
if there is no use case, then we dont need Enrich for .NET framework, right? (thats the current behavior)

Reply:
Just because I cannot see a usecase for the 3rd parameter. That doesn't mean there is no usecase at all. E.g. i want the activity.Displayname to be the same no matter what the underlaying db name is.

Reply:
to change activity.DisplayName, you can use ActivityProcessor. The whole Enrich feature was provided to provide easy access to raw objects from which users can extract additional information. In case of SQL on .NET FW, the raw object(payload) do not have anything not already parsed out by default.